src/ui/screens/WebcamCaptureTool.tsx

```diff
@@ -63,36 +66,46 @@ export default class WebcamCaptureTool extends React.Component<any,any> {
         };
     }
 
-  handleUploadPhoto = (dataUri: string) => {
-      let photo: PhotoAttach = {
-          data: dataUri,
-          encoding: 'base64',
-          type: 'image/png'
-      };
-
-      try {
-          photo = this.determinePhotoType(dataUri);
-      } catch (e) {
-          console.error("Couldn't parse the dataUri string to determine encoding type. Moving forward with the assumption that it is a PNG base64 string...")
-      } finally {
-          this.setState({
-              "photo~attach": photo
-          });
-      }
-  }
+    // Takes a data URI and returns the Data URI corresponding to the resized image at the wanted size.
+    resizeDataURL = async (datas: PhotoAttach) => {
+        return new Promise(async (resolve, reject) => {
+            var img = document.createElement('img');
+            img.onload = function()
+            {
+                var canvas = document.createElement('canvas');
+                var ctx: any = canvas.getContext('2d');
+                canvas.width = PHOTO_WIDTH;
+                canvas.height = PHOTO_HEIGHT;
+                try {
+                    ctx.drawImage(this, 0, 0, PHOTO_WIDTH, PHOTO_HEIGHT);
+                    var dataURI = canvas.toDataURL();
+                    resolve(dataURI);
+                } catch {
+                    reject();
+                }
+
+            };
+            img.src = `data:${datas.type};${datas.encoding},${datas.data}`
+        })
+    }
 
-  handleTakePhoto = (dataUri: string) => {
-      // The Camera component only ever uses PNGs, so we'll use that assumption and hardcode values accordingly
-      const data: string = dataUri.replace("data:image/png;base64,","")
-      const photo: PhotoAttach = {
-          data,
-          type: 'image/png',
-          encoding: 'base64'
-      };
-
-      this.setState({
-          "photo~attach": photo
-      });
+  savePhoto = async (dataUri: string) => {
+    let photo: PhotoAttach = {
+        data: dataUri,
+        encoding: 'base64',
+        type: 'image/png'
+    };
+
+    try {
+        photo = this.determinePhotoType(dataUri);
+    } catch (e) {
+        console.error("Couldn't parse the dataUri string to determine encoding type. Moving forward with the assumption that it is a PNG base64 string...")
+    } finally {
+        const resizedPhoto = await this.resizeDataURL(photo);
```

Comment:
I understand why you've added this variable to this block, but I'm wondering if it would be better to just create a `let` variable in the same scope as the `photo` variable and then set the `resizedPhoto` in the `try` block. Just thinking how we might catch the `reject()` from the method.